dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -436,8 +453,10 @@ internal open class RumViewScope(
                 isActive = !stopped,
                 cpuTicksCount = cpuTicks,
                 cpuTicksPerSecond = cpuTicks?.let { (it * ONE_SECOND_NS) / updatedDurationNs },
-                memoryAverage = lastMemoryVitalInfo?.meanValue,
-                memoryMax = lastMemoryVitalInfo?.maxValue
+                memoryAverage = lastMemoryInfo?.meanValue,
+                memoryMax = lastMemoryInfo?.maxValue,
+                refreshRateAverage = lastFrameRateInfo?.meanValue?.let { it * refreshRateScale },
```

Comment:
do we already have an idea how it will be displayed in the dashboard? It may be a bit confusing for the customer. Say I have an app where each frame takes 15 ms to render, so with 60 Hz mode I can get 60 fps. Then user has an option to switch refresh rate to 90 Hz in device settings and they do it (it may be switched by the user to save battery, or by the system dynamically). It means that we have now ~11 ms per frame, so now I'm a bit slow, so I skip some frames, but still I'm able to deliver 60 fps, but this time it will be normalized and become 40 fps. And for device for 60/90 Hz support they won't know in which mode this value is.

Maybe it is worth to send display refresh rate with event as well?

Reply:
What we get is the device capabilities, or more specifically what capabilities the window for the current activity has. 
If we're doing 60FPS instead of 90, whatever the reason we're still slow. The 90Hz option actually increase battery usage, and it's in the developer options anyway, so it won't be used that much I think. Also if the refresh rate is forced to 90Hz, why would it skip frames ?

Reply:
> Also if the refresh rate is forced to 90Hz, why would it skip frames ?

Because if frame construction pipeline is 15 ms, then it won't skip any frames with 60 Hz, but will start skipping with 90 Hz, because frame time is now different. Refresh rate can be adaptive and it can be controlled by the system https://www.gsmarena.com/newscomm-48261.php without any activity restart I guess. Plenty of devices produced by Oppo, Huawei, Xiaomi can do that.

Anyway, here my point is that we shouldn't probably do this normalization on the client side, but send device refresh rate to the backend with the metric we recorded. This is because with normalization it will add a confusion for the end user: say it was reported as 40 FPS (60 frames with 90 Hz capability). Then customer may think that his app is quite slow (is it 40 fps for 60 Hz limit? or 60 fps for 90 Hz limit? he won't know that), while in reality 60 fps on 90 Hz display is more or less acceptable and perception for the end user is not the same as for 40 fps. Customer may spend unnecessary dev effort because of this confusion.